bm.el

```diff
@@ -633,8 +686,16 @@ when `bm-next' or `bm-previous' navigate to this bookmark."
         (progn (setq bm-current bookmark)
                (overlay-put bookmark 'position (point-marker))
                (overlay-put bookmark 'time (or time (float-time))))
-      (let ((bookmark (make-overlay (bm-start-position) (bm-end-position)))
-            (hlface (if bm-buffer-persistence bm-persistent-face bm-face)))
+      (let ((hlface (if bm-buffer-persistence bm-persistent-face bm-face))
+            bookmark)
+        (when (and (not annotation) bm-annotate-on-create)
+          (setq annotation
+            (read-from-minibuffer "Annotation: "
```

Comment:
This code is not calling bm-bookmark-annotate() but instead duplicating parts of that code.